src/commands/positive.ts

```diff
@@ -65,184 +55,117 @@ export default class Positive extends ApiKeyCommand {
-            );
-            this.operationIdToResponseAndValidation[operationId][
-              parameterGroupName
-            ].validationFailures = failures;
-          });
-      },
-    );
+      for (const exampleGroup of exampleGroups) {
+        const operationExampleId = `${operation.getOperationId()}:${exampleGroup.getName()}`;
+        this.operationExamples.push({
+          id: operationExampleId,
+          operation,
+          exampleGroup,
+        });
+      }
+    }
   };
 
-  executeRequests = (): Promise<void>[] =>
-    this.operationIds
-      .filter((operationId) => {
-        // filter out all the requests that failed parameter validation
-        const existingValidations = this.operationIdToResponseAndValidation[
-          operationId
-        ];
-
-        return this.failedValidationGuard(existingValidations);
-      })
-      .flatMap((operationId) => {
-        const operationParameters = this.operationIdToParameters[operationId];
-
-        // If multiple parameter sets are present (due to example groups), execute once for each
-        if (Array.isArray(operationParameters)) {
-          return operationParameters
-            .filter((parameterExamples) => {
-              const groupName = Object.keys(parameterExamples)[0];
-              const existingValidations = this
-                .operationIdToResponseAndValidation[operationId];
-
-              return this.failedValidationGuard(existingValidations, groupName);
-            })
-            .map(async (parameterExamples) => {
-              const response = await this.executeRequest(
-                parameterExamples,
-                operationId,
-              );
-              const groupName = Object.keys(parameterExamples)[0];
+  validateParameters = (): void => {
+    for (const { id, exampleGroup } of this.operationExamples) {
+      const validator = new ParameterValidator(exampleGroup);
+      validator.validate();
 
-              if (!this.operationIdToResponseAndValidation[operationId]) {
-                this.operationIdToResponseAndValidation[operationId] = {};
-              }
+      this.operationFailures[id] = validator.getFailures();
+    }
+  };
 
-              this.operationIdToResponseAndValidation[operationId][
-                groupName
-              ] = {
-                response,
-              };
-            });
+  validateResponses = (responses: OperationResponse): void => {
+    for (const { id, operation } of this.operationExamples) {
+      const response = responses[id];
+      if (response) {
+        if (response.ok) {
+          const validator = new ResponseValidator(operation, response);
+          validator.validate();
+
+          this.operationFailures[id] = validator.getFailures();
+        } else {
+          this.operationFailures[id] = [
+            new ValidationFailure(
+              'Response status code was a non 2XX value',
+              [],
+            ),
+          ];
         }
+      }
```

Comment:
this would result in us adding a validation failure even if the parameters fail validation and the request is never run, so i think we (unfortunately) need to keep these checks separate.

Reply:
I'm having trouble understanding how these two are different, would you mind helping me understand your reply?
```js
if (res) {
  if (res.ok) {
    // if there is a response and it's okay, do this
  } else {
    // if there is a response and it's not okay do this
  }
}
```
```js
if (res?.ok) {
  // if there is a response and it's okay, do this
} else if (res) {
  // if there is a response and it's not okay do this
}
```

Reply:
oh lol, u right. I missed the update to the else statement.